groups/bdl/bdlb/bdlb_pcgrandomgenerator.h

```diff
@@ -0,0 +1,297 @@
+//      Die a(123, 456);  // BAD IDEA
+//
+//      int d1 = a.roll();
+//      int d2 = b.roll();
+//      assert(d2 == d1);
+//
+//  }
+//..
+///Example 2: Generating a random 32-bit number
+///- - - - - - - - - - - - - - - - - - - - - - -
+// The PCGRandomGenerator constructor takes two 64-bit constants (the initial
+// state, and the rng sequence selector.  Instances of 'PCGRandomGenerator' with
+// different sequence selectors will never have random sequences that coincide.
+// In order to obtain nondeterministic output for each instance, on a Unix
+// system random bytes can be obtained from '/dev/urandom'.  The component 
+// bdlb_randomdevice provides such functionality.  Also illustrated below is a 
+// a backup mechanism for obtaining random bytes.
+//..
+//  uint64_t state;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&state,
+//                                                         sizeof(state))) {
+//          state = time(NULL) ^ (intptr_t)&bsl::printf;  // fallback state
+//  }
+//  uint64_t streamSelector;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&streamSelector,
+//                                              sizeof(streamSelector))) {
+//           streamSelector = time(NULL) ^
+//                   (intptr_t)&bsl::printf;  // fallback streamSelector
+//  }
+//  bdlb::PCGRandomGenerator  rng(state, streamSelector);
+//  bsl::uint32_t randomInt = rng.generate();
+//..
+
+#include <bsl_cstdint.h>
+
+namespace BloombergLP {
+namespace bdlb {
+
+                                 // ========================
+                                 // class PCGRandomGenerator
+                                 // ========================
+class PCGRandomGenerator {
+    // This mechanism class implements a random number generator (RNG) based on
+    // the PCG algorithm.  PCG stands for "permuted congruential generator."
+    // The state is 64 bits.  It uses a so-called stream selector, also 64
+    // bits.  'initState' is the starting state for the RNG.  Any
+    // 64-bit value may be passed. 'streamSelector' selects the output sequence
+    // for the RNG.  Any 64-bit value may be passed, although only the low 63
+    // bits are significant.  There are 2^63 different RNGs available, and
+    // 'streamSelector' selects from among them.  Invoking different instances
+    // with the identical 'initState' and 'streamSelector' will result in the
+    // same sequence of random numbers from subsequent invocations of
+    // getRandom().  For details of the algorithm,see
+    // http://www.pcg-random.org.
+
+  private:
+    // DATA
+    bsl::uint64_t d_state;           // the RNG state
+
+    bsl::uint64_t d_streamSelector;  // selected sequence (stream)
+
+    // FRIENDS
+    friend bool operator==(const PCGRandomGenerator& lhs,
+                           const PCGRandomGenerator& rhs);
+
+  public:
+    // CREATORS
+    PCGRandomGenerator(bsl::uint64_t initState      = 0,
```

Comment:
What are 'initState' and 'streamSelector'?  they are not explained what they do or control.  

Further, there needs to be some discussion in this component (probably in whatever usage example we choose to show first) how these values should typically be chosen.   This should probably show both how one might want to deterministically select them, and how one might want to generate them randomly from bdlb_RandomDevice (or whatever).  

In summary: explain what `initState` and `streamSelector` do in the function doc here, then have the function doc refer to a usage example showing how someone might use these values